### PR TITLE
feat(acumen_product_agent): ignore skus

### PR DESCRIPTION
**Pull request by Jhalter5Stones**
ignore skus

### CODE REVIEW

**Comment by Jhalter5Stones** [comment]
@mrogers-5s Ready for review. The goal of this feature is to ignore a list of skus given in the options.

https://trello.com/c/Cjdq2f3S/49-usbat-donate

File: lib/huginn_acumen_product_agent/acumen_product_agent.rb [hunk 1]
```diff
@@ -54,6 +54,10 @@ def validate_options
             unless options['contributor_types_map'].is_a?(Hash)
                 errors.add(:base, "if provided, contributor_types_map must be a hash")
             end
+
+            unless options['ignore_skus'].is_a?(Array)
+                errors.add(:base, "if provided, ignore_skus must be an array")
+            end
```
Comment (mrogers-5s) [inline lib/huginn_acumen_product_agent/acumen_product_agent.rb:60]: I don't think this allows null values. It needs to.

`ignore_skus` should also default to an empty array when not provided.

File: lib/huginn_acumen_product_agent/acumen_product_agent.rb [hunk 2]
```diff
@@ -104,7 +109,19 @@ def handle(event)
...
+                  emit_product = true
+                  ignore_skus.each do |ignore_sku|
+                    if (ignore_sku == product['sku'])
+                      emit_product = false
+                    end
+                  end
+                  if (emit_product)
+                    create_event payload: product
+                  end
+                end
```
Comment (mrogers-5s) [inline lib/huginn_acumen_product_agent/acumen_product_agent.rb:124]:
```
unless ignore_skus.include?(product['sku'])
    create_event payload: product
end
```

**Review by mrogers-5s — CHANGES_REQUESTED** [review]

File: lib/huginn_acumen_product_agent/acumen_product_agent.rb (hunk 1 above)
Comment (Jhalter5Stones) [reply lib/huginn_acumen_product_agent/acumen_product_agent.rb]: @mrogers-5s This should be fixed. Let me know what you think

File: lib/huginn_acumen_product_agent/acumen_product_agent.rb (hunk 2 above)
Comment (Jhalter5Stones) [reply lib/huginn_acumen_product_agent/acumen_product_agent.rb]: @mrogers-5s This should be fixed let me know what you think.

**Review by mrogers-5s — APPROVED** [review]

**Comment by Jhalter5Stones** [comment]
@Ken-Michalak Ready for final review

https://trello.com/c/Cjdq2f3S/49-usbat-donate

**Review by Ken-Michalak — APPROVED** [review]

**Comment by Jhalter5Stones** [comment]
@jspizziri Can you release to ruby gems? Is there a way in a cli we can set something up to get released to ruby gems on merge for this project?

**Comment by jspizziri** [comment]
@Jhalter5Stones released as `v1.6.0`.

**Pull request merged**